### ISSUE TITLE
Decide future of "mesh-smooth" component

### ISSUE THREAD

**Issue by diarmidmackenzie** [c-frame/aframe-extras#411]
We've been reviewing all the components in this repo, and retiring a few that no longer seem valuable - see #395.

It's not clear what to do with 'mesh-smooth'.

As per [this comment](https://github.com/c-frame/aframe-extras/issues/395), it seems to still be compatible with the latest three.js, but we have no examples showing what it does, and from my testing I've not been able to get it to do much of any use.

When applied to a gltf, it seems to either have no effect, or (in one case) make the whole mesh go black with no materials - not very helpful.  See [this glitch example.
](https://mesh-smooth-component.glitch.me/)

Looking back to when it was introduced, it seems it was primarily tested with json models.
https://github.com/c-frame/aframe-extras/pull/165

The original proposal was a parameter on the json-model loader.
https://github.com/c-frame/aframe-extras/pull/163

That PR was rejected, and instead a generic component was proposed on the basis that "this would work for any file format, too".  But is it actually useful for any other fomat?

The JSON model loader has subsequently [been removed](https://github.com/c-frame/aframe-extras/commit/d079064e6ac55a4cd6bbf64bd46a576e26dd214e)

Only pause for hesitation before deleting the 'mesh-smooth' component is this comment from @donmccurdy at the end of #163 ["Actually I just ran into a use case for this with a glTF model, too."](https://github.com/c-frame/aframe-extras/pull/163#issuecomment-318372933) - though no detail on what this use case was...

It's such a simple component, I don't see any real loss from removing it, and I'd rather not clutter the repo with stuff where we don't understand the purpose or have any working examples.  If that causes anyone any problems, they can explain their use case & we can re-instate.

@vincentfretin - I propose we remove it as per[ your previous suggestion](https://github.com/c-frame/aframe-extras/issues/395#issuecomment-1375273210).  If you're OK with that I'll do a PR.

**Comment by vincentfretin** [comment]
Let's remove it for 7.0.0. If someone depends on this component, they can copy the previous code in their project or contribute a better one as a separate repository.

**Comment by donmccurdy** [comment]
Just some background — Knowing more now than I did then, it's true that the `mesh-smooth` component is format-agnostic. It doesn't generalize to all (most?) models though, because it relies on an assumption that vertices have already been welded. A somewhat more robust implementation would:

1. remove existing normals 
2. [weld vertices](https://threejs.org/docs/?q=buffergeom#examples/en/utils/BufferGeometryUtils.mergeVertices) with some error tolerance
3. compute vertex normals as before

That's a more expensive set of steps, and will still fail if vertices on adjacent faces have UV seams or vertex colors that prevent welding.

tl;dr — I think it's totally reasonable to remove the component, and I support that and any other reorganization you all choose to do here!

**Comment by vincentfretin** [comment]
Thanks @donmccurdy for your feedback!
